extension/sigv4authextension/config.go

```diff
@@ -44,6 +44,10 @@ var _ config.Extension = (*Config)(nil)
 // We aim to catch most errors here to ensure that we
 // fail early and to avoid revalidating static data.
 func (cfg *Config) Validate() error {
```

Comment:
Validate should not modify the configuration. Since you are already doing that below, I am not going to block merging this PR because of it, but I would like you to create an issue to move this and the `credsProvider` line below to `Unmarshal`

Reply:
Will take care of this early next week!